connectors/sources/github.py

```diff
@@ -775,44 +760,55 @@ async def get_github_item(self, resource):
         except Exception:
             raise
 
-    async def paginated_api_call(self, variables, query, keys):
+    async def paginated_api_call(self, query, variables, keys):
         """Make a paginated API call for fetching GitHub objects.
 
         Args:
-            variables (dict): Variables for Graphql API
             query (string): Graphql Query
+            variables (dict): Variables for Graphql API
             keys (list): List of fields to get pageInfo
 
         Yields:
             dict: dictionary containing response of GitHub.
         """
         while True:
-            query_data = {"query": query, "variables": variables}
-            response = await self.post(query_data=query_data)
+            response = await self.graphql(query=query, variables=variables)
             yield response
 
-            page_info = nested_get_from_dict(
-                response, ["data"] + keys + ["pageInfo"], default={}
-            )
+            page_info = nested_get_from_dict(response, keys + ["pageInfo"], default={})
             if not page_info.get("hasNextPage"):
                 break
             variables["cursor"] = page_info["endCursor"]  # pyright: ignore
 
     def get_repo_details(self, repo_name):
         return repo_name.split("/")
 
+    async def get_personal_access_token_scopes(self):
+        request_headers = sansio.create_headers(
+            self._get_client.requester,
+            accept=sansio.accept_format(),
+            oauth_token=self.github_token,
+        )
+        _, headers, _ = await self._get_client._request(
+            "HEAD", self.github_url, request_headers
+        )
+        scopes = headers.get("X-OAuth-Scopes")
+        if not scopes or not scopes.strip():
+            return set()
```

Comment:
Should this log a warning?

Reply:
Will add in the next PR

Reply:
Added in PR https://github.com/elastic/connectors/pull/2381

https://github.com/elastic/connectors/blob/75e8e51c610ba1da8e180ed486dafe89ebc7a5ca/connectors/sources/github.py#L859